File: 435-non-overlapping-intervals/non-overlapping-intervals.cpp

bool cmp(vector<int> a,vector<int> b){
  return a[1]<b[1];
}

class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(),intervals.end(),cmp);
        int count=0;
        int min=INT_MIN;
        for(int i=0;i<intervals.size();i++){
          if(intervals[i][0]>=min) min=intervals[i][1];
          else count++;
        }
        return count;
    }
};

File: 435-non-overlapping-intervals/non-overlapping-intervals.java
class Solution {
    public int eraseOverlapIntervals(int[][] intervals) {
        Arrays.sort(intervals,(a,b) -> a[1]-b[1]);

        int k=Integer.MIN_VALUE;
        int count=0;

        for(int i=0;i<intervals.length;i++){
            int start=intervals[i][0];
            int end=intervals[i][1];

            if(start>=k) k=end;
            else count++;
        }
        return count;
    }
}